cpp/src/io/csv/csv_reader.cu

```diff
@@ -1067,85 +1064,25 @@ __global__ void convertCsvToGdf(
-						int8_t *gdf_out = (int8_t *)gdf_data[actual_col];
-						gdf_out[rec_id] = convertStrtoInt<int8_t>(raw_csv, start, tempPos, opts.thousands);
-					}
-						break;
-					case gdf_dtype::GDF_INT16: {
-						int16_t *gdf_out = (int16_t *)gdf_data[actual_col];
-						gdf_out[rec_id] = convertStrtoInt<int16_t>(raw_csv, start, tempPos, opts.thousands);
-					}
-						break;
-					case gdf_dtype::GDF_INT32:
-					{
-						int32_t *gdf_out = (int32_t *)gdf_data[actual_col];
-						gdf_out[rec_id] = convertStrtoInt<int32_t>(raw_csv, start, tempPos, opts.thousands);
-					}
-						break;
-					case gdf_dtype::GDF_INT64:
-					{
-						int64_t *gdf_out = (int64_t *)gdf_data[actual_col];
-						gdf_out[rec_id] = convertStrtoInt<int64_t>(raw_csv, start, tempPos, opts.thousands);
-					}
-						break;
-					case gdf_dtype::GDF_FLOAT32:
-					{
-						float *gdf_out = (float *)gdf_data[actual_col];
-						gdf_out[rec_id] = convertStrtoFloat<float>(raw_csv, start, tempPos, opts.decimal, opts.thousands);
-					}
-						break;
-					case gdf_dtype::GDF_FLOAT64:
-					{
-						double *gdf_out = (double *)gdf_data[actual_col];
-						gdf_out[rec_id] = convertStrtoFloat<double>(raw_csv, start, tempPos, opts.decimal, opts.thousands);
-					}
-						break;
-					case gdf_dtype::GDF_DATE32:
-					{
-						gdf_date32 *gdf_out = (gdf_date32 *)gdf_data[actual_col];
-						gdf_out[rec_id] = parseDateFormat(raw_csv, start, tempPos, dayfirst);
-					}
-						break;
-					case gdf_dtype::GDF_DATE64:
-					{
-						gdf_date64 *gdf_out = (gdf_date64 *)gdf_data[actual_col];
-						gdf_out[rec_id] = parseDateTimeFormat(raw_csv, start, tempPos, dayfirst);
-					}
-						break;
-					case gdf_dtype::GDF_TIMESTAMP:
-					{
-						int64_t *gdf_out = (int64_t *)gdf_data[actual_col];
-						gdf_out[rec_id] = convertStrtoInt<int64_t>(raw_csv, start, tempPos, opts.thousands);
-					}
-					break;
-					case gdf_dtype::GDF_CATEGORY:
-					{
-						gdf_category *gdf_out = (gdf_category *)gdf_data[actual_col];
-						gdf_out[rec_id] = convertStrtoHash(raw_csv, start, pos, HASH_SEED);
-					}
-						break;
-					case gdf_dtype::GDF_STRING:
-					{
-						long end = pos;
-						if(opts.keepquotes==false){
-							if((raw_csv[start] == opts.quotechar) && (raw_csv[end-1] == opts.quotechar)){
-								start++;
-								end--;
-							}
+				// Type dispatcher does not handle GDF_STRINGS
+				if (dtype[col] == gdf_dtype::GDF_STRING) {
+					long end = pos;
+					if(opts.keepquotes==false){
+						if((raw_csv[start] == opts.quotechar) && (raw_csv[end-1] == opts.quotechar)){
+							start++;
+							end--;
 						}
-						str_cols[stringCol][rec_id].first	= raw_csv+start;
-						str_cols[stringCol][rec_id].second	= size_t(end-start);
-						stringCol++;
 					}
-						break;
-					default:
-						break;
+					str_cols[stringCol][rec_id].first	= raw_csv+start;
+					str_cols[stringCol][rec_id].second	= size_t(end-start);
+					stringCol++;
+				}
+				else {
+					cudf::type_dispatcher(
```

Comment:
Love it. Hopefully we'll get strings figured out soon so that we can also dispatch for `GDF_STRING`. In the mean time, your workaround is a good one.